### DISCUSSION TITLE
Integrate deeper into Vercel

### DISCUSSION THREAD

**Discussion by abriginets** [planetscale/discussion#97] [Feature Requests] [answered]
Hello. I have an idea how you guys could make Vercel integration even better. What do you think of functionality that will push the connection string directly into the Vercel's environment variables for different environments? Let's say I'm working on some feature and I'd like to create two databases for development and preview environments and put connection strings into Vercel's env variables but I'm to lazy to do it manually 😛

**Comment by shaunchurch** [comment, marked as answer]
I'd like to see this too. The existing environment variables do seem to have all the information required, but assembling it for use can be a problem in some environments. Prisma seems to require a single DATABASE_URL env var, and has no way to concatenate the existing variables, so I'm creating a full connection string manually at present. 

If anyone knows another way around this I'd be glad to hear it.

**Reply by abriginets** [reply]
When you generate a new password, you can select "Prisma" option and copy full DB url from there, but it's kinda silly anyway.

**Reply by janpio** [reply]
Prisma supports an `.env` file where you can "expand" variables - https://www.prisma.io/docs/guides/development-environment/environment-variables#expanding-variables Not sure if the Vercel integration offers all the necessary pieces to _construct_ the connection string - but if they do, this could be your solution.

**Reply by jgreet** [reply]
All of the pieces should be there to create that expanded database URL. Something like:

```
DATABASE_URL=mysql://${PLANETSCALE_DB_USERNAME}:${PLANETSCALE_DB_PASSWORD}@${PLANETSCALE_DB_HOST}/${PLANETSCALE_DB}?sslmode=require&sslaccept=strict&sslcert=${PLANETSCALE_SSL_CERT_PATH}
```

The intent here is that by providing the credentials in pieces we allow people to put together a connection string valid for whatever library or ORM they want to use on Vercel, but perhaps that's something we can handle in the integration in the future.

**Reply by NuroDev** [reply]
> Prisma seems to require a single DATABASE_URL env var, and has no way to concatenate the existing variables, so I'm creating a full connection string manually at present.

This is what I am currently doing. I am manually checking the `NODE_ENV` for a production build & overriding the datasource url. It technically does work but is not an elegant solution.

I have tried the above method & it does work when deploying manually with the expanded `DATABASE_URL` above. But for automated deployments I am yet to find a solution as providing that expanded environment variable as a Vercel project environment does not work as it doesn't get expanded.
